pennylane_qrack/qrack_device.cpp

```diff
@@ -757,44 +757,58 @@ struct QrackDevice final : public Catalyst::Runtime::QuantumDevice {
         std::copy(_p.get(), _p.get() + p.size(), p.begin());
 #endif
     }
+    void _SampleBody(const size_t numQubits, const std::map<bitCapInt, int>& q_samples, DataView<double, 2> &samples)
+    {
+        auto samplesIter = samples.begin();
+        auto q_samplesIter = q_samples.begin();
+        for (auto q_samplesIter = q_samples.begin(); q_samplesIter != q_samples.end(); ++q_samplesIter) {
+            const bitCapInt sample = q_samplesIter->first;
+            int shots = q_samplesIter->second;
+            for (; shots > 0; --shots) {
+                for (size_t wire = 0U; wire < numQubits; ++wire) {
+                    *(samplesIter++) = bi_compare_0((sample >> wire) & 1U) ? 1.0 : 0.0;
+                }
+            }
+        }
+    }
     void Sample(DataView<double, 2> &samples, size_t shots) override
     {
         // TODO: We could suggest, for upstream, that "shots" is a redundant parameter
```

Comment:
Do we want to do anything about this TODO (that is, do we want to create a specific issue and tag the reference number here for tacking?)

Reply:
We want to remove the "TODO," because it is "TODONE": https://github.com/PennyLaneAI/catalyst/issues/781